File: app/src/main/java/com/assignment/presentation/viewmodels/common/BaseViewModel.java
package com.assignment.presentation.viewmodels.common;

import android.arch.lifecycle.ViewModel;

import com.assignment.data.StatusData;


public class BaseViewModel extends ViewModel {

    private SingleLiveEvent<StatusData> mSingleLiveEvent;


    protected BaseViewModel(SingleLiveEvent<StatusData> singleLiveEvent) {
        this.mSingleLiveEvent = new SingleLiveEvent<>();
    }


    public SingleLiveEvent<StatusData> getSingleLiveEvent() {
        return mSingleLiveEvent;
    }
}

File: app/src/main/java/com/assignment/presentation/viewmodels/common/BaseViewModel.kt
package com.assignment.presentation.viewmodels.common

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.assignment.common.AssignmentApp
import com.assignment.presentation.helpers.IInternetStatus
import com.assignment.usecase.Intractors
import com.model.StatusData


open class BaseViewModel protected constructor(application: Application,
                                               intractors: Intractors, singleLiveEvent: SingleLiveEvent<StatusData>, mInternetStatus: IInternetStatus) : AndroidViewModel(application) {
    val singleLiveEvent: SingleLiveEvent<StatusData> =SingleLiveEvent()
    protected val application:AssignmentApp = getApplication()
}
